subsys/bluetooth/crypto/Kconfig

```diff
@@ -2,9 +2,23 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BT_CRYPTO
+	bool "Select crypto backend for Bluetooth Cryptographic Toolbox"
+	select BT_CRYPTO_PSA if MBEDTLS
+	select BT_CRYPTO_TINYCRYPT if !MBEDTLS
+	help
+	  This option enables the Bluetooth Cryptographic Toolbox.
+
+config BT_CRYPTO_PSA
```

Comment:
Thinking out loud. How about putting `BT_CRYPTO_PSA` and `BT_CRYPTO_TINYCRYPT` under a choice, that itself could be guarded by an `if BT_CRYPTO`?

Reply:
> There being a prompt or not doesn't change whether the user needs to manually select anything.

Yes, but if the user wants to take advantage of Mbed TLS it's enough to enable `CONFIG_MBEDTLS` in his/her build (I guess that most of the times Mbed TLS is used also for other purposes, so it's just there ready to be used).
`CONFIG_BT_CRYPTO` was not user-selectable before and it is not now; so do the new options that I added for the crypto backend: they are enabled based on what crypto provider is available in the system. This seems to be the most natural continuation of what was available before. Wdyt?

> Removing the prompt just prevents (to some extent) the user from assigning the Kconfig option.

Actually removing the prompt does not allow the user to select something even in `conf` file.

> And regarding the prompt, I think that you could still keep the particular choice options promptless

I tried, but that seems not possible.

Reply:
> CONFIG_BT_CRYPTO was not user-selectable before and it is not now; so do the new options that I added for the crypto backend: they are enabled based on what crypto provider is available in the system. This seems to be the most natural continuation of what was available before. Wdyt?

Yeah, I think it's good this way.

> Actually removing the prompt does not allow the user to select something even in conf file.

Here do you mean in `*.conf` files? If so that's the expected effect of promptless symbols. :slightly_smiling_face: 


Regarding the `choice` option (which is really the only thing I am interested about here). Is it so that you cannot make a choice promptless, which is the reason why you want to stick to individual options?

Reply:
> Regarding the choice option (which is really the only thing I am interested about here). Is it so that you cannot make a choice promptless, which is the reason why you want to stick to individual options?

Yes, it seems that it's not possible to make the choice promptless. I cannot complain though as I agree on the fact that `choice` means/requires user interaction (or at least the possibility to decide), which is what I'm not providing here.

Reply:
Okay. Having now tested things a bit locally, I get why you did what you did. This keeps the use of `BT_CRYPTO` identical to before. Less breaking changes can be good.